src/SupportView.ts

```diff
@@ -393,3 +395,99 @@ export function transposeMatrixIfNeeded(rowtype: IDType, d: IDataType) {
 
   return d;
 }
+
+
+async function splitMatrixInVectors(matrix: IAnyMatrix) {
+  const colNames = await matrix.cols();
+  const cols = matrix.ncol;
+  const r : IAnyVector[] = [];
+  for (let i = 0; i < cols; ++i) {
+    const v = matrix.slice(i);
+    const anyDesc : any = v.desc;
+    // hack the name to include the column label
+    anyDesc.name = matrix.desc.name + '/' + colNames[i];
+    anyDesc.origin = matrix;
+    anyDesc.fqname = matrix.desc.fqname + '/' + colNames[i];
+    r.push(v);
+  }
+  return r;
+}
+
+function convertToTree(datasets: IDataType[]) {
+  // has a origin field, i.e. is derived from a matrix
+  const isDerivedDataset = (dataset: IDataType) => !!(<any>dataset.desc).origin;
```

Comment:
! ... check for not existence
!! ... check for existence and ensure it is a boolean